src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    Node<K, V>[] table;
+    int threshold;
+    private int size;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (table != null) {
+            Node<K, V> head;
+            Node<K, V> returnValue = null;
+            Node<K, V>[] tab = table;
+            head = tab[Math.abs(hash(key) % tab.length)];
+            while (head != null) {
+                if ((key == head.key && key == null) || (key != null && key.equals(head.key))) {
+                    returnValue = head;
+                    break;
+                }
+                head = head.next;
+            }
+            return returnValue.value;
+        } else {
+            return null;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putValue(int hash, K key, V value) {
+        Node<K, V>[] tab = table;
+        int elementIndex;
+        if (tab == null || tab.length == 0) {
+            tab = resize();
+        }
+        elementIndex = Math.abs(hash(key) % tab.length);
+        if (tab[elementIndex] == null) {
+            tab[elementIndex] = new Node<>(hash, key, value, null);
+            size++;
+        } else {
+            Node<K, V> head = tab[elementIndex];
+            while (head != null) {
+                if (Objects.equals(head.key, key)) {
+                    head.value = value;
+                    break;
+                }
+                if (head.next == null) {
+                    head.next = new Node<>(hash, key, value, null);
+                    size++;
+                    break;
+                }
+                head = head.next;
+            }
+        }
+        if (size > threshold) {
+            resize();
+        }
+    }
+
+    private Node<K, V>[] resize() {
```

Comment:
do we need to return new array? let's operate with our table and make this method void